File: android/app/src/main/java/com/albbiz/map/ui/screens/JobsScreen.kt
// Bismillah Hir Rahman Nir Raheem
package com.albbiz.map.ui.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.ArrowBack
import androidx.compose.material.icons.filled.Work
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.lifecycle.viewmodel.compose.viewModel
import com.albbiz.map.data.Business
import com.albbiz.map.data.JobPosting
import com.albbiz.map.ui.LocalAppStrings
import com.albbiz.map.ui.MeTontGrey
import com.albbiz.map.ui.MeTontRed

// Same accent used for job postings on a business's own detail page
// (BusinessDetailsScreen's DetailJobItem) — kept consistent here since this
// screen is just every business's postings gathered into one place.
private val JobAccent = Color(0xFF673AB7)

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun JobsScreen(
    onBackClick: () -> Unit,
    onBusinessClick: (String) -> Unit,
    viewModel: MapViewModel = viewModel()
) {
    val businesses by viewModel.businesses.collectAsState()
    val strings = LocalAppStrings.current

    // Every business's job postings, flattened into one list, newest first —
    // this screen is the app-wide job board; a single business's own postings
    // still show on its own detail page too (DetailJobItem), this just
    // aggregates all of them in one browsable place.
    val allJobs = remember(businesses) {
        businesses
            .flatMap { business -> business.jobs.map { job -> business to job } }
            .sortedByDescending { (_, job) -> job.postedAt }
    }

    Scaffold(
        topBar = {
            TopAppBar(
                title = {
                    Text(
                        strings.jobs,
                        color = Color.White,
                        fontWeight = FontWeight.Bold
                    )
                },
                navigationIcon = {
                    IconButton(onClick = onBackClick) {
                        Icon(
                            Icons.AutoMirrored.Filled.ArrowBack,
                            "Back",
                            tint = Color.White
                        )
                    }
                },
                colors = TopAppBarDefaults.topAppBarColors(
                    containerColor = MeTontRed
                )
            )
        }
    ) { padding ->
        Box(
            modifier = Modifier
                .fillMaxSize()
                .background(MaterialTheme.colorScheme.background)
                .padding(padding)
        ) {
            if (allJobs.isEmpty()) {
                // ── EMPTY STATE ───────────────────────────────────
                Box(
                    modifier = Modifier.fillMaxSize(),
                    contentAlignment = Alignment.Center
                ) {
                    Column(
                        horizontalAlignment = Alignment.CenterHorizontally,
                        verticalArrangement = Arrangement.spacedBy(12.dp)
                    ) {
                        Surface(
                            modifier = Modifier.size(80.dp),
                            shape = RoundedCornerShape(40.dp),
                            color = JobAccent.copy(alpha = 0.1f)
                        ) {
                            Box(contentAlignment = Alignment.Center) {
                                Icon(
                                    Icons.Default.Work,
                                    null,
                                    modifier = Modifier.size(40.dp),
                                    tint = JobAccent.copy(alpha = 0.5f)
                                )
                            }
                        }
                        Text(
                            "No job postings yet",
                            style = MaterialTheme.typography.bodyLarge,
                            color = MeTontGrey,
                            fontWeight = FontWeight.Medium
                        )
                        Text(
                            "Businesses that are hiring will show up here",
                            style = MaterialTheme.typography.bodySmall,
                            color = MeTontGrey.copy(alpha = 0.7f)
                        )
                    }
                }
            } else {
                LazyColumn(
                    modifier = Modifier.fillMaxSize(),
                    contentPadding = PaddingValues(16.dp),
                    verticalArrangement = Arrangement.spacedBy(12.dp)
                ) {
                    item {
                        Text(
                            "${allJobs.size} open ${if (allJobs.size == 1) "position" else "positions"}",
                            color = MeTontGrey,
                            fontSize = 13.sp,
                            modifier = Modifier.padding(bottom = 4.dp)
                        )
                    }
                    items(allJobs) { (business, job) ->
                        JobListingCard(
                            business = business,
                            job = job,
                            onViewProfileClick = { onBusinessClick(business.id) }
                        )
                    }
                }
            }
        }
    }
}

@Composable
private fun JobListingCard(
    business: Business,
    job: JobPosting,
    onViewProfileClick: () -> Unit
) {
    Surface(
        shape = RoundedCornerShape(16.dp),
        shadowElevation = 2.dp,
        color = Color.White
    ) {
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .padding(16.dp),
            verticalArrangement = Arrangement.spacedBy(8.dp)
        ) {
            // ── BUSINESS NAME / CATEGORY ──────────────────────────
            Row(
                verticalAlignment = Alignment.CenterVertically,
                horizontalArrangement = Arrangement.spacedBy(8.dp)
            ) {
                Surface(
                    modifier = Modifier.size(8.dp),
                    shape = RoundedCornerShape(4.dp),
                    color = MeTontRed
                ) {}
                Text(
                    business.name,
                    style = MaterialTheme.typography.labelLarge,
                    fontWeight = FontWeight.SemiBold,
                    color = MeTontRed
                )
                if (business.category.isNotBlank()) {
                    Text(
                        "• ${business.category}",
                        style = MaterialTheme.typography.labelMedium,
                        color = MeTontGrey
                    )
                }
            }

            HorizontalDivider(color = Color(0xFFF0F0F0))

            // ── JOB TITLE ──────────────────────────────────────────
            Text(
                job.title,
                style = MaterialTheme.typography.titleMedium,
                fontWeight = FontWeight.Bold,
                color = Color.Black
            )

            // ── TYPE / SALARY ──────────────────────────────────────
            Row(
                verticalAlignment = Alignment.CenterVertically,
                horizontalArrangement = Arrangement.spacedBy(8.dp)
            ) {
                Surface(
                    shape = RoundedCornerShape(6.dp),
                    color = JobAccent.copy(alpha = 0.12f)
                ) {
                    Text(
                        job.type,
                        modifier = Modifier.padding(horizontal = 8.dp, vertical = 4.dp),
                        style = MaterialTheme.typography.labelSmall,
                        color = JobAccent,
                        fontWeight = FontWeight.Bold
                    )
                }
                if (job.salary != null) {
                    Text(
                        "💰 ${job.salary}",
                        style = MaterialTheme.typography.bodySmall,
                        color = MeTontGrey,
                        fontWeight = FontWeight.Medium
                    )
                }
            }

            // ── DESCRIPTION ────────────────────────────────────────
            Text(
                job.description,
                style = MaterialTheme.typography.bodySmall,
                color = MeTontGrey,
                maxLines = 3,
                overflow = TextOverflow.Ellipsis
            )

            Spacer(modifier = Modifier.height(4.dp))

            // ── VIEW PROFILE BUTTON ────────────────────────────────
            Button(
                onClick = onViewProfileClick,
                modifier = Modifier.fillMaxWidth(),
                shape = RoundedCornerShape(12.dp),
                colors = ButtonDefaults.buttonColors(
                    containerColor = MeTontRed,
                    contentColor = Color.White
                )
            ) {
                Text("View Profile", fontWeight = FontWeight.Bold)
            }
        }
    }
}

File: ios/AlbBizMap_IOS/AlbBizMap/UI/Screens/JobsScreen.swift
// Bismillah Hir Rahman Nir Raheem
import SwiftUI

// Same accent used for job postings on a business's own detail page —
// kept consistent here since this screen just gathers every business's
// postings into one place (matches Android's JobAccent).
private let jobAccent = Color(red: 0x67 / 255.0, green: 0x3A / 255.0, blue: 0xB7 / 255.0)

public struct JobsScreen: View {
    @Environment(\.appStrings) private var strings

    @ObservedObject public var viewModel: MapViewModel
    public let onBackClick: () -> Void
    public let onBusinessClick: (String) -> Void

    public init(viewModel: MapViewModel, onBackClick: @escaping () -> Void, onBusinessClick: @escaping (String) -> Void) {
        self.viewModel = viewModel
        self.onBackClick = onBackClick
        self.onBusinessClick = onBusinessClick
    }

    private var allJobs: [(Business, JobPosting)] {
        viewModel.businesses
            .flatMap { biz in biz.jobs.map { (biz, $0) } }
            .sorted { $0.1.postedAt > $1.1.postedAt }
    }

    public var body: some View {
        VStack(spacing: 0) {
            topAppBar

            if allJobs.isEmpty {
                emptyState
            } else {
                ScrollView {
                    VStack(alignment: .leading, spacing: 12) {
                        Text("\(allJobs.count) open \(allJobs.count == 1 ? "position" : "positions")")
                            .font(.system(size: 13))
                            .foregroundColor(.meTontGrey)

                        ForEach(allJobs, id: \.1.id) { biz, job in
                            JobListingCard(
                                business: biz,
                                job: job,
                                onViewProfileClick: { onBusinessClick(biz.id) }
                            )
                        }
                    }
                    .padding(16)
                }
            }
        }
        .background(Color.meTontBackground)
    }

    private var topAppBar: some View {
        HStack(spacing: 8) {
            Button(action: onBackClick) {
                Image(systemName: "chevron.left").foregroundColor(.white)
            }
            Text(strings.jobs)
                .font(.system(size: 18, weight: .bold))
                .foregroundColor(.white)
            Spacer()
        }
        .padding(.horizontal, 16)
        .padding(.vertical, 12)
        .background(Color.meTontRed)
    }

    private var emptyState: some View {
        VStack(spacing: 12) {
            ZStack {
                RoundedRectangle(cornerRadius: 40)
                    .fill(jobAccent.opacity(0.1))
                    .frame(width: 80, height: 80)
                Image(systemName: "briefcase.fill")
                    .font(.system(size: 34))
                    .foregroundColor(jobAccent.opacity(0.5))
            }
            Text(strings.jobsEmptyTitle)
                .font(.system(size: 16, weight: .medium))
                .foregroundColor(.meTontGrey)
            Text(strings.jobsEmptySubtitle)
                .font(.system(size: 13))
                .foregroundColor(.meTontGrey.opacity(0.7))
        }
        .frame(maxWidth: .infinity, maxHeight: .infinity)
    }
}

private struct JobListingCard: View {
    @Environment(\.appStrings) private var strings
    let business: Business
    let job: JobPosting
    let onViewProfileClick: () -> Void

    var body: some View {
        VStack(alignment: .leading, spacing: 8) {
            // Business name / category
            HStack(spacing: 8) {
                RoundedRectangle(cornerRadius: 4)
                    .fill(Color.meTontRed)
                    .frame(width: 8, height: 8)
                Text(business.name)
                    .font(.system(size: 14, weight: .semibold))
                    .foregroundColor(.meTontRed)
                if !business.category.trimmingCharacters(in: .whitespaces).isEmpty {
                    Text("• \(BusinessCategory.displayName(for: business.category))")
                        .font(.system(size: 12))
                        .foregroundColor(.meTontGrey)
                }
            }

            Divider().background(Color(red: 0xF0/255.0, green: 0xF0/255.0, blue: 0xF0/255.0))

            // Job title
            Text(job.title)
                .font(.system(size: 17, weight: .bold))
                .foregroundColor(.black)

            // Type / salary
            HStack(spacing: 8) {
                Text(job.type)
                    .font(.system(size: 11, weight: .bold))
                    .foregroundColor(jobAccent)
                    .padding(.horizontal, 8)
                    .padding(.vertical, 4)
                    .background(jobAccent.opacity(0.12))
                    .cornerRadius(6)

                if let salary = job.salary, !salary.isEmpty {
                    Text("💰 \(salary)")
                        .font(.system(size: 13, weight: .medium))
                        .foregroundColor(.meTontGrey)
                }
            }

            // Description
            Text(job.description)
                .font(.system(size: 13))
                .foregroundColor(.meTontGrey)
                .lineLimit(3)

            Spacer().frame(height: 4)

            // View Profile button
            Button(action: onViewProfileClick) {
                Text(strings.viewProfile)
                    .fontWeight(.bold)
                    .frame(maxWidth: .infinity)
                    .frame(height: 44)
                    .background(Color.meTontRed)
                    .foregroundColor(.white)
                    .cornerRadius(12)
            }
        }
        .padding(16)
        .background(Color.white)
        .cornerRadius(16)
        .shadow(color: Color.black.opacity(0.06), radius: 4, x: 0, y: 2)
    }
}
